File: programmers/java/부족한_금액_계산하기.java
// https://programmers.co.kr/learn/courses/30/lessons/82612

class Solution {
    public long solution(int price, int money, int count) {
        long answer = -1;
        long need_money = 0;
        
        for(int i=1; i<count+1; i++)
            need_money += i * price;
        
        if(need_money > money)
            return (need_money - money);
        else
            return 0;
    }
}

File: programmers/python/부족한_금액_계산하기.py
# https://programmers.co.kr/learn/courses/30/lessons/82612?language=python3
def solution(price, money, count):
    need_money = 0
    
    for i in range(1, count+1):
        need_money += i*price
        
    if need_money <= money:
        return 0
    else:
        return need_money - money
    